server/vulnerabilities/db.go

```diff
@@ -0,0 +1,159 @@
+package vulnerabilities
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/facebookincubator/nvdtools/cpedict"
+	"github.com/facebookincubator/nvdtools/wfn"
+	"github.com/jmoiron/sqlx"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+func CPEDB(dbPath string) (*sqlx.DB, error) {
+	db, err := sqlx.Open("sqlite3", dbPath)
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+func applyCPEDatabaseSchema(dbPath string) error {
+	db, err := CPEDB(dbPath)
+	if err != nil {
+		return err
+	}
+	_, err = db.Exec(`
+	CREATE TABLE IF NOT EXISTS cpe (
+		cpe23 TEXT NOT NULL,
+		title TEXT NOT NULL,
+		version TEXT,
+		target_sw TEXT,
+		deprecated BOOLEAN DEFAULT FALSE
+	);
+	CREATE TABLE IF NOT EXISTS deprecated_by (
+		cpe_id INTEGER,
+		cpe23 TEXT NOT NULL,
+		FOREIGN KEY(cpe_id) REFERENCES cpe(rowid)
+	);
+	CREATE VIRTUAL TABLE IF NOT EXISTS cpe_search USING fts5(title, target_sw);
+	CREATE INDEX IF NOT EXISTS idx_version ON cpe (version);
+	CREATE INDEX IF NOT EXISTS idx_cpe23 ON cpe (cpe23);
+	CREATE INDEX IF NOT EXISTS idx_target_sw ON cpe (target_sw);
+	CREATE INDEX IF NOT EXISTS idx_deprecated_by ON deprecated_by (cpe23);
+	`)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func InsertCPEItem(db *sqlx.DB, item cpedict.CPEItem) ([]interface{}, map[string]string, error) {
```

Comment:
`db` seems to be unused? Does this need to be exported? Maybe this should be called `generateCPEItem`?